File: app/src/main/java/zachwhaley/atwork/AtWorkLocationReceiver.java
package zachwhaley.atwork;

import android.content.Context;
import android.content.Intent;
import android.support.v4.content.WakefulBroadcastReceiver;

public class AtWorkLocationReceiver extends WakefulBroadcastReceiver {

    @Override
    public void onReceive(Context context, Intent intent) {
        AtWork.log("AtWorkLocationReceiver.onReceive");
        Intent locationService = new Intent(context, AtWorkLocationService.class);
        startWakefulService(context, locationService);
    }
}

File: app/src/main/java/zachwhaley/atwork/AtWorkLocationReceiver.kt
package zachwhaley.atwork

import android.content.Context
import android.content.Intent
import android.support.v4.content.WakefulBroadcastReceiver

class AtWorkLocationReceiver : WakefulBroadcastReceiver() {

    override fun onReceive(context: Context, intent: Intent) {
        AtWork.log("AtWorkLocationReceiver.onReceive")
        val locationService = Intent(context, AtWorkLocationService::class.java)
        WakefulBroadcastReceiver.startWakefulService(context, locationService)
    }
}
